.travis.yml

```diff
@@ -1,55 +1,55 @@
+  - ANDROID_BUILD_TOOLS_VERSION=28.0.3
+  - ANDROID_ABI=armeabi-v7a
 
 android:
   components:
-    # Uncomment the lines below if you want to
-    # use the latest revision of Android SDK Tools
     - tools
     - platform-tools
-
-    # The BuildTools version used by your project
-    - build-tools-28.0.3
-
-    # The SDK version used to compile your project
-    - android-28
+    - tools
+    - extra-android-m2repository
+  licenses:
+    - 'android-sdk-preview-license-52d11cd2'
+    - 'android-sdk-license-.+'
+    - 'google-gdk-license-.+'
+
+before_install:
+  - touch $HOME/.android/repositories.cfg
+  - yes | sdkmanager "platforms;android-28"
+  - yes | sdkmanager "build-tools;28.0.3"
+  - yes | sdkmanager "ndk-bundle"
 
 before_cache:
-  - rm -fr $HOME/.gradle/caches
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
 
 cache:
   directories:
-# Travis is not happy with gradle cache (getting permission denied)
-#   - $HOME/.gradle/caches/
+  - $HOME/.gradle/caches/
+  - $HOME/.gradle/wrapper/
+  - $HOME/.android/build-cache
 
-# Build with the development SDK
 before_script:
-  - /bin/sh ./set_debug_env.sh
+  - chmod +x gradlew
 
 # Just build the project for now
 script:
-  # Build app (assembleAppRelease assembleAppfdroidRelease)
-  # Build Android test (assembleAndroidTest)
-  # Code quality (lintAppRelease lintAppfdroidRelease)
-  - ./gradlew clean assembleAppRelease assembleAppfdroidRelease assembleAndroidTest lintAppRelease lintAppfdroidRelease --stacktrace
-  # Run unitary test (Disable for now, see https://travis-ci.org/vector-im/riot-android/builds/502504370)
-  # - ./gradlew testAppReleaseUnitTest --stacktrace
-  # Other code quality check
-  - ./tools/check/check_code_quality.sh
-  - ./tools/travis/check_pr.sh
-  # Check that indonesians file are identical. Due to Android issue, the resource folder must be value-in/, and Weblate export data into value-id/.
-  - diff ./vector/src/main/res/values-id/strings.xml ./vector/src/main/res/values-in/strings.xml
+ - ./gradlew assembleAppDebug
+ 
+deploy:
+  provider: releases
+  file: "vector/build/outputs/apk/app/debug/vector-app-debug.apk"
+  skip_cleanup: true
+  draft: true
+  on:
+    branch: patch-1
+  api_key:
+    secure: "qh/v61K0ZP/C29FG57JC3sPeVlJw3D2VXA+M5sym+GMR5mOQd3RkofB4kdSwma0zj6AiEVaTBZ+VWnadLBAOW9avz7BqS2MUn8EbU8dAunwuHiyoMwfq4hPofolt7ftpOO5PaBb3pBYcJdld/94ai+yix11IpxShvM/li5AS+XnAYYne4NQ82GedftPl4pdMsbtla0CBD3lmg1GZzJNALZUvvoxE9lXwkRfPT7bq6pGM+yHkCrCYpIjIyuSCGi5bV6d5uAz8XPXL6C20Gd/kaI47OrkyMvVMZJwCQMSlycwOtiPIv6hWX4v8yb1NQ7NtlHOREM/JwMaDhyeCLwLP+uGDfshA96xD+9z4glySYbxn50orc/DRD6QkfBJW219KTcMRHtueR7VZZl0K1vMjmEzioP6mabyrsZE96fxxJAMBfwgsrozb/gHtHRimoHcs3R4rNu8CLgjolKTesdkhSvbSjasUotKqP9OXE+8kSqojToRQurmnRqcYTpORlOi1/8DTaEoXZu4fDKgOwod2N3fvxK5sxp66yF9yWC+76yR0UMU8sL/ZXVUoO5q3Duteqqk5GytQkUREsQO46cPaUEJY3/tBS9YCeTcPiaM4veOaWZGEkds4jPbromUpa/GFhj8/vEmC4QYZ/U/u9tU8a7IgDwqd5GaWDCuGED5YvfY="
```

Comment:
What key is this and where is this getting deployed?

Reply:
From https://medium.com/@harmittaa/travis-ci-android-example-357f6e632fc4

You can generate the API key by going to your account settings on GitHub, Personal access tokens, Generate new token. Set the scope to public_repo and generate the token. Remember to copy the access token.

Encrypt the API key (you will need to install travis gem locally: gem install travis) and add it to the deploy section under api_key.

```travis encrypt apikey```

Reply:
Cool. I will merge this PR, but I am going to knock out the deploy part, as I am not comfortable putting keys out there in the cloud.